docs/dev_guide/ceph_s3_tests/ceph_s3_tests_guide.md

```diff
@@ -0,0 +1,249 @@
+# Ceph S3 Tests Guide
+## Introduction
+* Ceph S3 tests are unofficial AWS S3 compatibility tests written in Python (for more information see: [ceph/s3-test](https://github.com/ceph/s3-tests) repository on GitHub) that we use in our CI.
+* This guide contains:
+     1) General Settings For Ceph S3 Tests
+     2) Run All Ceph S3 Tests
+     3) Run a Single Ceph S3 Test
+     4) Debug a Single Ceph S3 Test
+     5) Examples
+* This guide describes developer steps to run Ceph S3 on a noobaa system on minikube.
+
+## General Settings For Ceph S3 Tests
+We assume that it is not your first deployment of noobaa system, and you already succeeded with it (If not, please see the guide Deploy Noobaa On Minikube).
+We will run the commands in the terminal, you may work with at least two tabs:
+1) For noobaa-core repository
+2) For noobaa-operator repository
+
+More tabs will be used to view the endpoint logs, connect to the tester pod, etc.
+In each step, it is mentioned what tab you should use.
+
+### 1) Before Building The Images (Noobaa-Core Tab)
+We will use minikube to run the tests. It is recommended to build all images on the minikube docker daemon. Configure your docker client to use minikube's docker run:
+```bash
+eval $(minikube docker-env)
+```
+
+### 2) Build Operator (Noobaa-Operator Tab)
+In order to build the CLI and the operator image run the following:
+```bash
+. ./devenv.sh
+make all
+```
+Note: the file `devenv.sh` contains the command `eval $(minikube docker-env)`. We run the command `eval $(minikube docker-env)` prior to an image build (whether from noobaa core repository or noobaa operator repository). 
+
+This will build the following:
+* noobaa-operator image with tag `noobaa/noobaa-operator:<major.minor.patch>` (for example: `noobaa/noobaa-operator:5.13.0`). this tag is used by default when installing with the CLI.
+* noobaa CLI. The `devenv.sh` script is setting an alias `nb` to run the local build of the CLI.
+
+### 3) Build Core And Tester Images (Noobaa-Core Tab)
+Run the following to build noobaa core image with the desired tag to build the tester image:
+```bash
+make tester TESTER_TAG=noobaa-tester:s3-tests
+docker tag noobaa:latest noobaa-core:s3-tests
+```
```

Comment:
If we are using a different tab for building the core image then the command `eval $(minikube docker-env)` should run on that tab too, otherwise core will be running in the system docker daemon.

Reply:
It was mentioned here `1) Before Building The Images (Noobaa-Core Tab)` (it is the same tab).

Reply:
Oh ok, missed it,